### PR TITLE
Add Battlepass pallet

### LINKED ISSUE #227: Create `battlepass` pallet
### What needs to be done
Create `Battlepass` pallet with basic support for battlepass functionality.

### Acceptance Criteria/Task List**
Next extrinsics should be created:
- [x] create_battlepass
- [x] claim_battlepass

#### create_battlepass - extrinsic to create battle pass for organization.
Parameters:
  * origin - should be one of: sudo, org prime
  * org_id - organization id to create battlepass for
  * price - price to pay for the user to get access to battle pass
  * cid - content id
  
Outcome:
Battlepass entity created onchain with given parameters (org_id, price, cide). Later this entity will be used to track battlepass progress.

#### claim_battlepass - extrinsic to get access to existing battlepass for the user, who wants to participate
Parameters:
  * origin - user which wants buys battlepass
  * org_id - organization id, which battlepass belongs to
  * battlepass_id - battlepass id
  * discord_id _[optional]_ - discord id for the user, if any
  * payment_id _[optional]_ - external id for the payment, made by the user
  
Outcome
NFT is generated and assigned to the user, granting access to the battlepass, it was requested for.
In future it will allow user to finish tasks, earn points and claim battlepass rewards with those points.

### PR DESCRIPTION
**Pull request by FiberMan**

### CODE REVIEW

File: bin/alphaville/runtime/Cargo.toml [hunk 1]
```diff
@@ -292,6 +294,7 @@ runtime-benchmarks = [
 	"gamedao-control/runtime-benchmarks",
 	"gamedao-flow/runtime-benchmarks",
 	"gamedao-signal/runtime-benchmarks",
+	"gamedao-battlepass/runtime-benchmarks",
```
Comment (vovacha) [inline bin/alphaville/runtime/Cargo.toml:297]: There are no benchmarks defined for this pallet.

File: bin/alphaville/runtime/src/lib.rs [hunk 2]
```diff
@@ -1799,6 +1799,16 @@ impl gamedao_sense::Config for Runtime {
...
 }
 
+impl gamedao_battlepass::Config for Runtime {
+	type Event = Event;
+	type Control = Control;
+	type Rmrk = RmrkCore;
+	type StringLimit = StringLimit;
+	type SymbolLimit = ConstU32<100>;
+	type PartsLimit = ConstU32<25>;
+	type MaxResourcesOnMint = ConstU32<100>;
```
Comment (vovacha) [inline bin/alphaville/runtime/src/lib.rs]: We can reuse the same constants for `rmrk_core` and this pallet with the `parameter_types!` macro.

**Review by vovacha — CHANGES_REQUESTED** [review]
LGTM, but building with benchmarks doesn't work (subzero-dev). Can you check it once more?
`cargo build --release --features runtime-benchmarks`

File: bin/alphaville/runtime/Cargo.toml (hunk 1 above)
Comment (FiberMan) [reply bin/alphaville/runtime/Cargo.toml:297]: Not yet, will add benchmarks in separate PR.

**Pull request merged**